tests/phpunit/tests/post/nav-menu.php

```diff
@@ -202,6 +202,81 @@ public function test_wp_get_nav_menu_items_with_taxonomy_term() {
 		$this->assertEquals( $t, $menu_items[0]->object_id );
 	}
 
+	/**
+	 * @ticket 55620
+	 */
+	public function test_update_menu_item_cache_primed_post() {
+		$post_id = self::factory()->post->create();
+		wp_update_nav_menu_item(
+			$this->menu_id,
+			0,
+			array(
+				'menu-item-type'      => 'post_type',
+				'menu-item-object'    => 'post',
+				'menu-item-object-id' => $post_id,
+				'menu-item-status'    => 'publish',
+			)
+		);
+
+		$posts_query  = new WP_Query();
+		$query_result = $posts_query->query( array( 'post_type' => 'nav_menu_item' ) );
+
+		wp_cache_delete( $post_id, 'posts' );
+		$action = new MockAction();
```

Comment:
@manuelRod Let's do the test like this - https://github.com/WordPress/wordpress-develop/pull/2612/files#diff-7ddf35c200bedaa7eb0205ccb94ee5e145f8132da934d4b3b06548e981d49adbR1786